tensorboard/uploader/flags_parser.py

```diff
@@ -0,0 +1,203 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Flags parser for TensorBoard.dev uploader."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from absl.flags import argparse_flags
+
+SUBCOMMAND_FLAG = "_uploader__subcommand"
+SUBCOMMAND_KEY_UPLOAD = "UPLOAD"
+SUBCOMMAND_KEY_DELETE = "DELETE"
+SUBCOMMAND_KEY_LIST = "LIST"
+SUBCOMMAND_KEY_EXPORT = "EXPORT"
+SUBCOMMAND_KEY_UPDATE_METADATA = "UPDATEMETADATA"
+SUBCOMMAND_KEY_AUTH = "AUTH"
+AUTH_SUBCOMMAND_FLAG = "_uploader__subcommand_auth"
+AUTH_SUBCOMMAND_KEY_REVOKE = "REVOKE"
+
+DEFAULT_ORIGIN = "https://tensorboard.dev"
+
+
+def parse_flags(argv):
```

Comment:
I can do it in a followup PR.  But first I'll warn the TensorBoard team that we'll be getting rid of //tensorboard/uploader just in case anybody is still using it.  And this function would essentially be moved into the test.